core/payment-driver/erc20/src/erc20/wallet.rs

```diff
@@ -141,11 +143,11 @@ pub async fn make_transfer(
     let amount = big_dec_to_u256(&amount_big_dec)?;
     let gas_price = match gas_price {
         Some(gas_price) => big_dec_gwei_to_u256(gas_price)?,
-        None => convert_float_gas_to_u256(POLYGON_STARTING_GAS_PRICE),
+        None => convert_float_gas_to_u256(get_polygon_starting_price()),
```

Comment:
When not providing `gas_price`, we're assuming that we're using Polygon (or using Polygon defaults). This should be matched against `network`